src/api/controllers/CHANNEL_PINS_UPDATE.ts

```diff
@@ -0,0 +1,18 @@
+import { eventHandlers } from "../../bot.ts";
+import {
+  DiscordChannelPinsUpdateEvent,
+  DiscordPayload,
+} from "../../types/mod.ts";
+import { cacheHandlers } from "./cache.ts";
+
+export async function handleChannelPinsUpdate(data: DiscordPayload) {
+  const payload = data.d as DiscordChannelPinsUpdateEvent;
+
+  const channel = await cacheHandlers.get("channels", payload.channel_id);
+  if (!channel) return;
+
+  const guild = await cacheHandlers.get("guilds", payload.guild_id ?? "");
```

Comment:
I think this is better than doing an invalid request
```suggestion
  const guild = payload.guild_id ? await cacheHandlers.get("guilds", payload.guild_id) : undefined;
```